src/engine/engine_common.c

```diff
@@ -432,13 +425,159 @@ static int lock_clines(struct ocf_request *req,
 	}
 }
 
+static inline int ocf_evict_part_overflow(struct ocf_request *req,
+		const struct ocf_engine_callbacks *engine_cbs, uint32_t clines_to_evict)
+{
+	int lock_status = -OCF_ERR_NO_LOCK;
+	struct ocf_metadata_lock *metadata_lock = &req->cache->metadata.lock;
+
+	ocf_metadata_start_exclusive_access(metadata_lock);
+
+	if (space_managment_evict_do(req->cache, req, clines_to_evict) ==
+			LOOKUP_MISS) {
+		goto unlock;
+	}
+
+	if (!ocf_part_is_enabled(req->cache, req->part_id)) {
+		ocf_req_set_mapping_error(req);
+		goto unlock;
+	}
+
+	lock_status = lock_clines(req, engine_cbs);
+
+unlock:
+	ocf_metadata_end_exclusive_access(metadata_lock);
+
+	return lock_status;
+}
+
+static inline int ocf_prepare_clines_hit(struct ocf_request *req,
+		const struct ocf_engine_callbacks *engine_cbs)
+{
+	int lock_status = -OCF_ERR_NO_LOCK;
+	struct ocf_metadata_lock *metadata_lock = &req->cache->metadata.lock;
+	uint32_t clines_to_evict;
+	int res;
+
+	/* Cachelines are mapped in correct partition */
+	if (ocf_part_is_enabled(req->cache, req->part_id) &&
+			!ocf_engine_needs_repart(req)) {
+		lock_status = lock_clines(req, engine_cbs);
+		ocf_req_hash_unlock_rd(req);
+		return lock_status;
+	}
+
+	res = ocf_part_check_space(req, &clines_to_evict);
+
+    /* Check whether target partition if full */
+	if (res == OCF_PART_HAS_SPACE)
+		lock_status = lock_clines(req, engine_cbs);
+
+	if (res == OCF_PART_IS_DISABLED)
+		ocf_req_set_mapping_error(req);
+
+	ocf_req_hash_unlock_rd(req);
+
+	if (res != OCF_PART_IS_FULL)
```

Comment:
if (res == OCF_PART_HAS_SPACE)

Reply:
This check is correct - if part would be disabled, but had cachelines assigned, `ocf_part_check_space()` return `OCF_PART_IS_FULL`. Any other status means that no eviction is needed. It also implies correctness of mapping error set in :477. I will add comment to my code

Reply:
got it